src/algebra/continued_fractions/computation/approximations.lean

```diff
@@ -0,0 +1,278 @@
+-/
+
+/-- Shows that the integer parts of the continued fraction are at least one. -/
+lemma of_one_le_nth_part_denom {b : K}
+  (nth_part_denom_eq : (gcf.of v).partial_denominators.nth n = some b) :
+  1 ≤ b :=
+begin
+  obtain ⟨gp_n,  nth_s_eq, ⟨refl⟩⟩ : ∃ gp_n, (gcf.of v).s.nth n = some gp_n ∧ gp_n.b = b, from
+    obtain_s_b_of_part_denom nth_part_denom_eq,
+  obtain ⟨ifp_n, succ_nth_stream_eq, ifp_n_b_eq_gp_n_b⟩ :
+    ∃ ifp, int_fract_pair.stream v (n + 1) = some ifp ∧ (ifp.b : K) = gp_n.b, from
+      int_fract_pair.obtain_succ_nth_stream_of_gcf_of_nth_eq_some nth_s_eq,
+  rw [←ifp_n_b_eq_gp_n_b],
+  exact_mod_cast (int_fract_pair.one_le_succ_nth_stream_b succ_nth_stream_eq)
+end
+
+/--
+Shows that the partial numerators `aᵢ` of the continued fraction are equal to one and the partial
+denominators `bᵢ` correspond to integers.
+-/
+lemma of_part_num_eq_one_and_part_denom_corr_int {gp : gcf.pair K}
+  (nth_s_eq : (gcf.of v).s.nth n = some gp) :
+  gp.a = 1 ∧ ∃ (z :ℤ), gp.b = (z : K) :=
+begin
+  obtain ⟨ifp, stream_succ_nth_eq, ifp_b_eq_gp_b⟩ :
+    ∃ ifp, int_fract_pair.stream v (n + 1) = some ifp ∧ (ifp.b : K) = gp.b,
+      from int_fract_pair.obtain_succ_nth_stream_of_gcf_of_nth_eq_some nth_s_eq,
+  have : (gcf.of v).s.nth n = some ⟨1, ifp.b⟩, from
+    nth_of_eq_some_of_succ_nth_int_fract_pair_stream stream_succ_nth_eq,
+  have : some gp = some ⟨1, ifp.b⟩, by rwa nth_s_eq at this,
+  have : gp = ⟨1, ifp.b⟩, by injection this,
+  -- We know the shape of gp, so now we just have to split the goal and use this knowledge.
+  cases gp,
+  split,
+  { injection this },
+  { existsi ifp.b, injection this }
+end
+
+/-- Shows that the partial numerators `aᵢ` are equal to one. -/
+lemma of_part_num_eq_one {a : K} (nth_part_num_eq : (gcf.of v).partial_numerators.nth n = some a) :
+  a = 1 :=
+begin
+  obtain ⟨gp, nth_s_eq, gp_a_eq_a_n⟩ : ∃ gp, (gcf.of v).s.nth n = some gp ∧ gp.a = a, from
+    obtain_s_a_of_part_num nth_part_num_eq,
+  have : gp.a = 1, from (of_part_num_eq_one_and_part_denom_corr_int nth_s_eq).left,
+  rwa gp_a_eq_a_n at this
+end
+
+/-- Shows that the partial denominators `bᵢ` correspond to an integer. -/
+lemma of_part_denom_corr_int {b : K}
+  (nth_part_denom_eq : (gcf.of v).partial_denominators.nth n = some b) :
+  ∃ (z : ℤ), b = (z : K) :=
+begin
+  obtain ⟨gp, nth_s_eq, gp_b_eq_b_n⟩ : ∃ gp, (gcf.of v).s.nth n = some gp ∧ gp.b = b, from
+    obtain_s_b_of_part_denom nth_part_denom_eq,
+  have : ∃ (z : ℤ), gp.b = (z : K), from (of_part_num_eq_one_and_part_denom_corr_int nth_s_eq).right,
+  rwa gp_b_eq_b_n at this
+end
+
+/-
+Our next goal is to show that `Bₙ₊₁ ≥ bₙ* Bₙ`. For this, we first show that the partial denominators
+`Bₙ` grow quicker than the fibonacci sequence `nat.fib`.
+-/
+
+-- open `nat` as we will make use of fibonacci numbers.
+open nat
+
+lemma of_fib_le_continuants_aux_b : (n ≤ 1 ∨ ¬(gcf.of v).terminated_at (n - 2)) →
+  (fib n : K) ≤ ((gcf.of v).continuants_aux n).b :=
+nat.strong_induction_on n
+(begin
```

Comment:
The parentheses around the begin end block are not needed, and this might look better with the first line of the proof being `refine nat.strong_induction_on n (\lam n IH hyp, _)` instead of doing it in term mode

Reply:
Fixed